### DISCUSSION TITLE
Database Connection configuration

### DISCUSSION THREAD

**Discussion by hussamitani** [spatie/laravel-medialibrary#2698] [Ideas] [answered]
Basically we are splitting our Laravel App into multiple Databases. We need to define the db connection in the Model. Please see the following documentation: https://medium.com/teknomuslim/how-to-run-laravel-using-multiple-database-connections-d7e8dcb7ed2a

When I added `protected $connection = 'mysql'` to the Media Model in vendor/spatie/laravel-medialibrary/src/MediaCollections/Models/Media it worked like a charm.

Can you add this option to the config/media-library.php? Because this can be published in our src config files.

Thank you very much

**Comment by erikn69** [comment, marked as answer]
Make your own model with `protected $connection = 'mysql'` and extends `Spatie\MediaLibrary\MediaCollections\Models\Media::class`
There is a config for custom properties, only you don't read the documentation
https://github.com/spatie/laravel-medialibrary/blob/a6634fad75e32d72e5af47719850a88121107428/config/media-library.php#L28-L31
>Remember, many hours of trial and error can save you 5 minutes of looking at the documentation.

**Reply by hussamitani** [reply]
Nice that is actually the perfect solution. I was already thinking of making my own model that extends Media but I didnt know how to tell the trait to use that Model. Thank you very much
